### DISCUSSION TITLE
Flash-relaying

### DISCUSSION THREAD

**Discussion by razgraf** [sablier-labs/v2-periphery#12] [Ideas]
During a recent discussion we touched upon relayers a.k.a. other people paying gas for someone's withdrawals. Up until now we were thinking of relayers as possible rescuers (#155), whitelisted by the Sablier contracts through a permissioned system. 

Why not split these two concepts and create a standalone, permissionless relayer system.

### Idea

Create a helper contract that implements a flash-relay method. For the lifetime of one block, the contract will
1. transfer the NFT from your wallet to itself (thus becoming the recipient)
2. withdraw funds from the stream
3. at the end transfer both the NFT and the funds back to your wallet. 
 
As a pre-requirement, you (the holder) would have to allow this contract to transfer the NFT. This way, someone running a "keeper" could become the caller of this relaying contract. Should we also add an embedded fee here?

**Comment by PaulRBerg** [comment]
Interesting idea. In practice, the way this would work is that we build a simple UI for recovering lost funds, UI which performs the necessary approvals and calls the appropriate smart contract in the backend.

I like this and the mere existence of this possibility makes me hesitant to march on with the rescuers (https://github.com/sablierhq/v2-core/discussions/155) proposal.

**Comment by PaulRBerg** [comment]
Closing as we have decided to make the withdraw function publicly callable:

https://github.com/sablier-labs/v2-core/issues/731
